Cig tweaks

comma opened the pull request:
Smokables now require proper gas to burn, any oxydizer will do.
They also consume it and produce trace amounts of CO2 and CO. Not really enough to be an issue in bigger rooms or with working scrubbers, but if you're being really stupid with limited space and air, it'll do something.
Fixes #10371. Similar check existed in helmet off code, but not for masks.
Removes light setting for cigs. It was just too huge, and our mangled light system doesn't allow subtle lights anymore. Ember is layered over darkness anyway.

<!-- 
Do not forget to add a changelog when you have made admin/player facing changes that can alter gameplay.
Examples which require a changelog entry include:
* Adding/removing objects that players may interact with, or the way they function.
* Adding/removing/altering admin tools.
* Changing the map.

Examples were changelog entries are optional/not typically required:
* Cosmetic changes such as descriptions, sound effects, etc.
* Optimizations and other changes to underlying systems which do not affect gameplay.
* Minor bug fixes.

You find a README and example file in .\html\changelogs\ for further instructions.
-->

Sbotkin commented:
Does this mean I won't able to smoke with my helmet on? Damn.
![image](https://user-images.githubusercontent.com/5092934/45923033-42a89c00-bee3-11e8-8006-cc73b387a64f.png)

comma commented:
Actually you can, it would use air from internals

Sbotkin commented:
Oh, okay, it's good then :+1:

BloodyMan commented:
Your internals would still shut off the moment the cig runs dry.

comma commented:
No, that's fixed in tis PR

comma committed:
Cig tweaks

Smokables now require proper gas to burn, any oxydizer will do. If internals are on, they'll use that instead (for in-helmet smoking).
They also consume it and produce trace amounts of CO2 and CO. Not really enough to be an issue in bigger rooms or with working scrubbers, but if you're being really stupid with limited space and air, it'll do something.
Fixes #10371. Similar check existed in helmet off code, but not for masks.
Removes light setting for cigs. It was just too huge, and our mangled light system doesn't allow subtle lights anymore. Ember is layered over darkness anyway.
